index.js

```diff
@@ -66,6 +66,14 @@ function csurf (options) {
   var ignoreMethod = getIgnoredMethods(ignoreMethods)
 
   return function csrf (req, res, next) {
+    if (req.isInitialized) {
+      console.warn('[Warning] csurf() is duplicately called with same middleware in the cooke mode, first validation will result in the invalid token.')
+    }
+
+    if (!req.isInitialized && cookie) {
+      req.isInitialized = true
```

Comment:
I think `req.isInitialized` is just too generic, unfortunately. Users and other middlewares can define properties on `req`, so it should be something that is more confined to csurf.

Reply:
Thank for reviewing, I pushed new code.